packages/salesforcedx-utils-vscode/src/helpers/traceFlags.ts

```diff
@@ -141,7 +142,13 @@ export class TraceFlags {
       'TraceFlag',
       traceFlag
     )) as DataRecordResult;
-    return result.success && result.id ? result.id : undefined;
+
+    if (result.success && result.id) {
+      TraceFlagsRemover.getInstance(this.connection).addNewTraceFlagId(result.id);
```

Comment:
If the user starts a new session, would this trace flag still be removed?

Reply:
By new session, do you if they quit VS Code and re-run?  `connection` is coming from `workspaceContext.getConnection()`.  I looked at `connection.accessToken` and it was the same value between runs, even after I quit VSC, ran VSC again, and then ran the command.